### PR TITLE
Add support for Windows credential manager.

### PR DESCRIPTION
**Pull request by calavera**
/cc @jhowardmsft

Signed-off-by: David Calavera <david.calavera@gmail.com>

### CODE REVIEW

**Comment by jhowardmsft** [comment]
Wow, nice! @jstarks

File: wincred/wincred_windows.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+type wincred struct{}
+
+// New creates a new wincred.
+func New() credentials.Helper {
+	return wincred{}
+}
+
+// Add adds new credentials to the windows credentials manager.
+func (h wincred) Add(creds *Credentials) error {
+	dp := winc.NewDomainPassword(creds.ServerURL)
```
Comment (jstarks) [inline wincred/wincred_windows.go]: I think you want to use NewGenericCredential and not NewDomainPassword. The latter is used automatically to connect to other Windows machines e.g. for SMB shares and stuff.

Comment (calavera) [inline wincred/wincred_windows.go]: thanks @jstarks. I fixed it already.

**Comment by calavera** [comment]
Tests pass on appveyor, so I'm going to assume that it works :laughing: Merging and moving on with the next one.

**Pull request merged**